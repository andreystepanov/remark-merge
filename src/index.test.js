import merge from './'
import remark from 'remark'

// prettier-ignore
const data = `
# Title

Content...

> Quote text 1

> Quote text 2

Some paragraph...

> Quote text 3

Another paragraph...

> Quote text 4

> Quote text 5

> Quote text 6

More paragraphs...

\`\`\`
const param1 = true
\`\`\`

\`\`\`
const param2 = true
\`\`\`

\`\`\`
const param3 = true
\`\`\`

Another paragraph

\`\`\`
const param4 = true
\`\`\`

\`\`\`
const param5 = true
\`\`\`

Anoter paragraph

\`\`\`
const end = true
\`\`\`

Last paragraph
`

function parse(data, options) {
  return remark()
    .use(merge, options)
    .use({ settings: { fences: true } })
    .processSync(data)
    .toString()
}

test('defined', () => {
  expect(merge).toBeDefined()
  expect(typeof merge).toBe('function')
})

test('ignores types that not in whitelist', () => {
  const markdown = parse(data, { types: ['image'] })
  expect(markdown).toMatchSnapshot()
})

test('merges default types', () => {
  const markdown = parse(data)
  expect(markdown).toMatchSnapshot()
})

test('merges only `code` blocks', () => {
  const markdown = parse(data, { types: ['code'] })
  expect(markdown).toMatchSnapshot()
})

test('ignores line breaks in `devider` option', () => {
  const markdown = parse(data, { devider: '---\n\n' })
  expect(markdown).toMatchSnapshot()
})
