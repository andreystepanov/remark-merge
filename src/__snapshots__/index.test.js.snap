// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ignores line breaks in \`devider\` option 1`] = `
"# Title

Content...

> Quote text 1
>
> ---
>
> Quote text 2

Some paragraph...

> Quote text 3

Another paragraph...

> Quote text 4
>
> ---
>
> Quote text 5
>
> ---
>
> Quote text 6

More paragraphs...

\`\`\`
const param1 = true

---

const param2 = true

---

const param3 = true
\`\`\`

Another paragraph

\`\`\`
const param4 = true

---

const param5 = true
\`\`\`

Anoter paragraph

\`\`\`
const end = true
\`\`\`

Last paragraph
"
`;

exports[`ignores types that not in whitelist 1`] = `
"# Title

Content...

> Quote text 1

> Quote text 2

Some paragraph...

> Quote text 3

Another paragraph...

> Quote text 4

> Quote text 5

> Quote text 6

More paragraphs...

\`\`\`
const param1 = true
\`\`\`

\`\`\`
const param2 = true
\`\`\`

\`\`\`
const param3 = true
\`\`\`

Another paragraph

\`\`\`
const param4 = true
\`\`\`

\`\`\`
const param5 = true
\`\`\`

Anoter paragraph

\`\`\`
const end = true
\`\`\`

Last paragraph
"
`;

exports[`merges default types 1`] = `
"# Title

Content...

> Quote text 1
>
> Quote text 2

Some paragraph...

> Quote text 3

Another paragraph...

> Quote text 4
>
> Quote text 5
>
> Quote text 6

More paragraphs...

\`\`\`
const param1 = true

const param2 = true

const param3 = true
\`\`\`

Another paragraph

\`\`\`
const param4 = true

const param5 = true
\`\`\`

Anoter paragraph

\`\`\`
const end = true
\`\`\`

Last paragraph
"
`;

exports[`merges only \`code\` blocks 1`] = `
"# Title

Content...

> Quote text 1

> Quote text 2

Some paragraph...

> Quote text 3

Another paragraph...

> Quote text 4

> Quote text 5

> Quote text 6

More paragraphs...

\`\`\`
const param1 = true

const param2 = true

const param3 = true
\`\`\`

Another paragraph

\`\`\`
const param4 = true

const param5 = true
\`\`\`

Anoter paragraph

\`\`\`
const end = true
\`\`\`

Last paragraph
"
`;
